src/amo/reducers/addonsByAuthors.js

```diff
@@ -4,27 +4,34 @@ import invariant from 'invariant';
 
 import { createInternalAddon } from 'core/reducers/addons';
 import type {
-  SearchResultAddonType, ExternalAddonType,
+  ExternalAddonType,
+  SearchResultAddonType,
 } from 'core/types/addons';
 
 
-type State = {
+type AddonId = number;
+
+export type AddonsByAuthorsState = {|
   // TODO: It might be nice to eventually stop storing add-ons in this
   // reducer at all and rely on the add-ons in the `addons` reducer.
   // That said, these are partial add-ons returned from the search
   // results and fetching all add-on data for each add-on might be too
   // expensive.
-  byAddonId: { [number]: Array<SearchResultAddonType> },
-  byAddonSlug: { [string]: Array<number> },
-  byUserId: { [number]: Array<number> },
-  byUsername: { [string]: Array<number> },
-};
+  byAddonId: { [AddonId]: SearchResultAddonType },
+  byAddonSlug: { [string]: Array<AddonId> },
+  byUserId: { [number]: Array<AddonId> },
+  byUsername: { [string]: Array<AddonId> },
+  forAuthorNamesAndAddonType: { [string]: Array<AddonId> | null },
```

Comment:
Because this is storing the add-ons found *for* a group of authorNames + addonType. But fair enough, I will change it because once I typed that I didn't buy it. 😉